op-acceptance-tests/tests/jovian/min_base_fee_test.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type minBaseFee struct {
+	// Ctx is the context for test execution.
+	ctx context.Context
+	// log is the component-specific logger instance.
+	log log.Logger
+	// T is a minimal test interface for panic-checks / assertions.
+	t devtest.T
+	// Require is a helper around the above T, ready to assert against.
+	require *testreq.Assertions
+
+	l1Client     *dsl.L1ELNode
+	l2Network    *dsl.L2Network
+	l2EL         *dsl.L2ELNode
+	systemConfig minBaseFeeSystemConfig
+
+	originalMinBaseFee uint64
+}
+
+type minBaseFeeSystemConfig struct {
+	SetMinBaseFee func(minBaseFee uint64) bindings.TypedCall[any] `sol:"setMinBaseFee"`
+	MinBaseFee    func() bindings.TypedCall[uint64]               `sol:"minBaseFee"`
+}
+
+func newMinBaseFee(t devtest.T, l2Network *dsl.L2Network, l1EL *dsl.L1ELNode, l2EL *dsl.L2ELNode) *minBaseFee {
+	systemConfig := bindings.NewBindings[minBaseFeeSystemConfig](
+		bindings.WithClient(l1EL.EthClient()),
+		bindings.WithTo(l2Network.Escape().Deployment().SystemConfigProxyAddr()),
+		bindings.WithTest(t))
+
+	originalMinBaseFee, err := contractio.Read(systemConfig.MinBaseFee(), t.Ctx())
+	t.Require().NoError(err, "reading original minBaseFee")
+
+	return &minBaseFee{
+		ctx:                t.Ctx(),
+		log:                t.Logger(),
+		t:                  t,
+		require:            t.Require(),
+		l1Client:           l1EL,
+		l2Network:          l2Network,
+		l2EL:               l2EL,
+		systemConfig:       systemConfig,
+		originalMinBaseFee: originalMinBaseFee,
+	}
+}
+
+func (mbf *minBaseFee) checkCompatibility() bool {
+	_, err := contractio.Read(mbf.systemConfig.MinBaseFee(), mbf.ctx)
+	if err != nil {
+		mbf.t.Fail()
+		return false
+	}
+	return true
+}
+
+func (mbf *minBaseFee) getSystemOwner() *dsl.EOA {
+	priv := mbf.l2Network.Escape().Keys().Secret(devkeys.SystemConfigOwner.Key(mbf.l2Network.ChainID().ToBig()))
+	return dsl.NewKey(mbf.t, priv).User(mbf.l1Client)
+}
+
+func (mbf *minBaseFee) setMinBaseFee(minBaseFee uint64) {
+	owner := mbf.getSystemOwner()
+
+	_, err := contractio.Write(mbf.systemConfig.SetMinBaseFee(minBaseFee), mbf.ctx, owner.Plan())
+	mbf.require.NoError(err, "SetMinBaseFee transaction failed")
+
+	mbf.t.Logf("Set min base fee on L1: minBaseFee=%d", minBaseFee)
+}
+
+func (mbf *minBaseFee) verifyMinBaseFee(from *dsl.EOA, to *dsl.EOA, minBase *big.Int, shouldEnforce bool) {
+	// Simulate user transactions
+	for range 20 {
+		from.Transfer(to.Address(), eth.OneGWei)
+	}
+
+	info := mbf.getBlock()
+	prevBlockNum := info.NumberU64()
+	for range 5 {
+		n := mbf.getBlock()
+		mbf.require.True(n.NumberU64() > prevBlockNum, "block number should increase")
+
+		prevBlockNum = n.NumberU64()
+		feeCmp := n.BaseFee().Cmp(minBase)
+		if !shouldEnforce {
+			mbf.require.True(feeCmp > 0, "expected base fee to be higher than the minBaseFee")
+		} else {
+			mbf.require.True(feeCmp == 0, "expected base fee to be at least minBaseFee")
```

Comment:
Bit confused by this. Did you mean to write `>0`? Semantically it doesn't seem super clear, because if we are "not enforcing" a minimum, one might think that the base fee could drop below that minimum. Because actually what we do is enforce a trivial minimum.

I'm thinking we should run 3 test cases. One with a moderate minimum, one with a very high minimum, and one with a zero minimum. Then we simply assert that the base fee is greater than or equal to the minimum in all cases after the required number of blocks.